File: Escola/Cursos.Site/app/services/ProvaquestoesService.js
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/13/2017 11:55:26 AM
///// Gerado automaticamente para alteracoes dos metodos comum alterar o template
///// </sumary>
"use strict";
app.factory("ProvaquestoesService", ["$http", "ngAuthSettings", function ($http, ngAuthSettings) {
        var service = {};
		var serviceBase = ngAuthSettings.apiServiceBaseUri;
        service.GetAll = function (callback) {
            $http.get(serviceBase + "Provaquestoes/getall")
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.GetById = function (id, callback) {
            $http.get(serviceBase + "Provaquestoes/getbyid/" + id)
               .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Post = function (Provaquestoes, callback) {
            $http.post(serviceBase + "Provaquestoes/post", Provaquestoes)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Put = function (Provaquestoes, callback) {
            $http.post(serviceBase + "Provaquestoes/put", Provaquestoes)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

		service.Delete = function (Provaquestoes, callback) {
            $http.post(serviceBase + "Provaquestoes/delete", Provaquestoes)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        return service;
    }]);

File: Escola/Cursos.Repository/Services/ProvaquestoesService.cs
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/11/2017 9:23:24 PM

///// Para outros metodos implementar classe partial
///// </sumary>
using System;
using System.Linq;
using System.Collections.Generic;
using System.Threading.Tasks;
using Cursos.Repository.Services.Base;
using Cursos.Data;
using Cursos.Repository.Services.Interfaces;
using System.Linq.Expressions;
using Cursos.Repository.Work.Interface;
namespace  Cursos.Repository.Services
{
	  /// <summary>
      /// Public Service Class  ProvaquestoesService
      /// </summary>
      public partial class ProvaquestoesService : BaseServices, IProvaquestoesService
      {

		public ProvaquestoesService(IUnitOfWork unitOfWork)
        {
			UnitOfWork = unitOfWork;
        }

		/// <summary>
        /// Add Range Entity EV_PROVAQUESTOES
        /// </summary>
        /// <param name="entity"></param>
		public void Add(IEnumerable<EV_PROVAQUESTOES> entity)
        {
            UnitOfWork.ProvaquestoesRepository.Add(entity);
        }

		/// <summary>
        /// Add Entity EV_PROVAQUESTOES
        /// </summary>
        /// <param name="entity"></param>
		public void Add(EV_PROVAQUESTOES entity)
        {	
            UnitOfWork.ProvaquestoesRepository.Add(entity);
        }
		/// <summary>
        /// Delete Entity EV_PROVAQUESTOES
        /// </summary>
        /// <param name="entity"></param>
        public void Delete(EV_PROVAQUESTOES entity)
        {
            UnitOfWork.ProvaquestoesRepository.Delete(entity);
        }
		 /// <summary>
        /// Get All Entity EV_PROVAQUESTOES
        /// </summary>
        /// <returns>IEnumerable EV_PROVAQUESTOES</returns>
        public IEnumerable<EV_PROVAQUESTOES> GetAll()
        {
			return UnitOfWork.ProvaquestoesRepository.GetAll();
        }
	     
		/// <summary>
        /// Update Entity EV_PROVAQUESTOES
        /// </summary>
        /// <param name="entity"></param>
        public void Update(EV_PROVAQUESTOES entity)
        {
            UnitOfWork.ProvaquestoesRepository.Edit(entity);
        }

		/// <summary>
        /// Commit Transaction
        /// </summary>
        /// <returns></returns>
        public int Commit()
        {
            return UnitOfWork.Commit();
        }
		/// <summary>
        /// Commit Async Transaction
        /// </summary>
		public Task<int> CommitAsync()
        {
            return UnitOfWork.CommitAsync();
        }
				/// <summary>
        /// Get Entity EV_PROVAQUESTOES By Id
        /// </summary>
        /// <param name="id"></param>
        /// <returns>EV_PROVAQUESTOES</returns>
		public EV_PROVAQUESTOES GetById(int id)
        {
			return UnitOfWork.ProvaquestoesRepository.FindBy(x => x.ID.Equals(id)).SingleOrDefault();
        }		
				
		public IEnumerable<EV_PROVAQUESTOES> FindBy(Expression<Func<EV_PROVAQUESTOES, bool>> predicate)
        {
            return UnitOfWork.ProvaquestoesRepository.FindBy(predicate);
        }
      }
}
